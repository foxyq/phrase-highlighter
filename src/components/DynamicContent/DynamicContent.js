import React from 'react';

const DynamicContent = props => {
  return <>{props.children}</>;
};

export default DynamicContent;
