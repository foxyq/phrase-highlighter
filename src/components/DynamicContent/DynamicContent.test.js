import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import DynamicContent from './DynamicContent';

describe('DynamicContent component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<DynamicContent />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  const content = 'abc';

  it('renders props correctly', () => {
    const Component = renderer
      .create(<DynamicContent>{content}</DynamicContent>)
      .toJSON();
    expect(Component).toMatchSnapshot();
  });

  it('renders no props correctly', () => {
    const Component = renderer.create(<DynamicContent />).toJSON();
    expect(Component).toMatchSnapshot();
  });
});
