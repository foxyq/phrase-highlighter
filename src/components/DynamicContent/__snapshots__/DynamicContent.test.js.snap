// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicContent component renders no props correctly 1`] = `null`;

exports[`DynamicContent component renders props correctly 1`] = `"abc"`;
